### PR TITLE
Improve instrumentation around Zombie Task detection.

### PR DESCRIPTION
**Pull request by spershin**
Summary:
- Change Driver::toString() to log more relevant info.
- Change Task::toString() to log more relevant info.
- Change Operator::toString() to log more relevant info.

Differential Revision: D56734557

### CODE REVIEW

File: velox/exec/Driver.cpp [hunk 1]
```diff
@@ -987,15 +990,30 @@ std::vector<Operator*> Driver::operators() const {
...
 std::string Driver::toString() const {
   std::stringstream out;
-  out << "{Driver: ";
-  if (state_.isOnThread()) {
+  out << "{Driver." << driverCtx()->pipelineId << "." << driverCtx()->driverId
+      << ": ";
+  if (state_.isTerminated) {
+    out << "terminated, ";
+  }
+  if (state_.hasBlockingFuture) {
```
Comment (xiaoxmeng) [inline velox/exec/Driver.cpp:998]: Shall we only print out block op if the driver is blocked?

File: velox/exec/Task.cpp [hunk 2]
```diff
@@ -2229,23 +2230,44 @@ ContinueFuture Task::taskCompletionFuture() {
 std::string Task::toString() const {
   std::lock_guard<std::timed_mutex> l(mutex_);
   std::stringstream out;
-  out << "{Task " << shortId(taskId_) << " (" << taskId_ << ")";
+  out << "{Task " << shortId(taskId_) << " (" << taskId_ << ")" << std::endl;
```
Comment (xiaoxmeng) [inline velox/exec/Task.cpp:2233]: We can tell from driver's reference count? Why we need to make shallow drivers? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@spershin thanks for the change % comments.

File: velox/exec/Task.cpp (hunk 2 above)
Comment (spershin) [reply velox/exec/Task.cpp:2233]: @xiaoxmeng 

Thanks for taking time to look at the change!
Hard to find a reviewer these days. 🙁

We need Driver shadows for few reasons:
1. When task is terminated it loses all references to the Drivers - meaning we can't find drivers that might still linger out there.
2. If there are Driver shadows - then we can log some details about these drivers.
3. If there are Driver shadows - then we know that it is them making the Task zombie and not something else.

Let me know if that explains the idea behind them.,

**Review by xiaoxmeng — APPROVED** [review]
@spershin thanks for the update!

**Pull request closed without merging**